src/control/server/mgmt_system.go

```diff
@@ -408,3 +414,43 @@ func (svc *mgmtSvc) StartRanks(ctx context.Context, req *mgmtpb.RanksReq) (*mgmt
 
 	return resp, nil
 }
+
+// ClusterEvent management service gRPC handler receives ClusterEvent requests
+// from control-plane instances attempting to notify the MS of a cluster event
+// in the DAOS system.
+//
+// On receipt of the cluster event request, process any encapsulated RAS event
+// messages (handling of other types to be supported in the future).
+//
+// Necessary state updates for the rank involved in the event will be
+// registered in the system membership.
+func (svc *mgmtSvc) ClusterEvent(ctx context.Context, req *mgmtpb.ClusterEventReq) (*mgmtpb.ClusterEventResp, error) {
+	if req == nil {
+		return nil, errors.New("nil request")
+	}
+	if req.Sequence < 1 {
+		return nil, errors.New("invalid sequence number in request")
+	}
+	svc.log.Debugf("MgmtSvc.ClusterEvent dispatch, req:%#v\n", req)
+
+	if err := svc.sysdb.CheckLeader(); err != nil {
+		return nil, err
+	}
+
+	rasEventPB := req.GetRas()
+	if rasEventPB == nil {
+		return nil, errors.Errorf("unexpected event type received, want RAS got %T",
+			req.GetEvent())
+	}
+
+	event, err := events.NewFromProto(rasEventPB)
+	if err != nil {
+		return nil, err
+	}
+	svc.dispatchEvents(event)
```

Comment:
Same question as above, what if this is unset?

Reply:
moved the eventPubSub into the mgmtSvc and enable events function called directly now